cog/core/sql_abstract.py

```
@@ -0,0 +1,188 @@
+
+        for field in fields(record):
+            if record.is_unset(field.name):
+                raise Exception(  # pylint: disable=broad-exception-raised
+                    f"SQL is not return all fields. (`{field.name}`=`{getattr(record, field.name)}`)"
+                )
+
+        return record
+
+    def to_sql(self):
+        if self._protected:
+            raise ValueError("You should new a object to modify.")
```

Ruff (0.8.2)

Avoid specifying long messages outside the exception class

(TRY003)